v1/providers/nebius/image.go

```diff
@@ -2,10 +2,293 @@ package v1
+			continue
+		}
+
+		img := v1.Image{
+			ID:           image.Metadata.Id,
+			Name:         image.Metadata.Name,
+			Description:  getImageDescription(image),
+			Architecture: extractArchitecture(image),
+		}
+
+		if image.Metadata.CreatedAt != nil {
+			img.CreatedAt = image.Metadata.CreatedAt.AsTime()
+		}
+
+		images = append(images, img)
+	}
+
+	return images, nil
+}
+
+// getRegionalPublicImages retrieves public images for the specified region
+func (c *NebiusClient) getRegionalPublicImages(ctx context.Context, region string) ([]v1.Image, error) {
+	// Determine the correct public images parent for this region
+	publicParent := c.getPublicImagesParentForRegion(region)
+
+	imagesResp, err := c.sdk.Services().Compute().V1().Image().List(ctx, &compute.ListImagesRequest{
+		ParentId: publicParent,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list public images for region %s: %w", region, err)
+	}
+
+	var images []v1.Image
+	for _, image := range imagesResp.GetItems() {
+		if image.Metadata == nil {
+			continue
+		}
+
+		img := v1.Image{
+			ID:           image.Metadata.Id,
+			Name:         image.Metadata.Name,
+			Description:  getImageDescription(image),
+			Architecture: extractArchitecture(image),
+		}
+
+		if image.Metadata.CreatedAt != nil {
+			img.CreatedAt = image.Metadata.CreatedAt.AsTime()
+		}
+
+		images = append(images, img)
+	}
+
+	return images, nil
+}
+
+// getCrossRegionPublicImages tries to get public images from other regions as fallback
+func (c *NebiusClient) getCrossRegionPublicImages(ctx context.Context) ([]v1.Image, error) {
+	// Common region patterns to try
+	regions := []string{"eu-north1", "eu-west1", "us-central1"}
+
+	for _, region := range regions {
+		if region == c.location {
+			continue // Skip current region since we already tried it
+		}
+
+		images, err := c.getRegionalPublicImages(ctx, region)
+		if err == nil && len(images) > 0 {
+			return images, nil // Return first successful region
+		}
+	}
+
+	return c.getDefaultImages(ctx) // Final fallback
+}
+
+// getPublicImagesParentForRegion determines the correct public images parent ID for a region
+func (c *NebiusClient) getPublicImagesParentForRegion(region string) string {
+	// Map region to routing code patterns
+	regionToRoutingCode := map[string]string{
+		"eu-north1":  "e00",
+		"eu-west1":   "e00",
+		"us-central1": "u00",
+		"us-west1":   "u00",
+		"asia-southeast1": "a00",
+	}
+
+	if routingCode, exists := regionToRoutingCode[region]; exists {
+		return fmt.Sprintf("project-%spublic-images", routingCode)
+	}
+
+	// Fallback: try to extract from current project ID
+	return c.getPublicImagesParent()
+}
+
+// getDefaultImages returns common public images when no project-specific images are found
+func (c *NebiusClient) getDefaultImages(ctx context.Context) ([]v1.Image, error) {
+	// Common Nebius public image families
+	defaultFamilies := []string{
+		"ubuntu22.04-cuda12",
+		"ubuntu20.04",
+		"ubuntu18.04",
```

Comment:
do we want to support such an old image? I feel like we should maybe drop this one and include 24 (if they offer it)?